frontend/appflowy_tauri/src/appflowy_app/components/document/BlockMenu/BlockMenu.hooks.ts

```diff
@@ -0,0 +1,105 @@
+import { DocumentControllerContext } from '$app/stores/effects/document/document_controller';
+import { documentActions, Node } from '@/appflowy_app/stores/reducers/document/slice';
+import { nanoid } from 'nanoid';
+import { useSubscribeNode } from '../_shared/SubscribeNode.hooks';
+import { useAppDispatch } from '@/appflowy_app/stores/store';
+import { useCallback, useContext, useRef, useState, useEffect } from 'react';
+import { BlockType } from '@/appflowy_app/interfaces/document';
+export function useBlockMenu(nodeId: string, open: boolean) {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const { insertAfter, remove } = useActions(nodeId);
+  const dispatch = useAppDispatch();
+  const [style, setStyle] = useState({ top: '0px', left: '0px' });
+
+  useEffect(() => {
+    if (!open) {
+      return;
+    }
+    // set selection when open
+    dispatch(documentActions.setSelectionById(nodeId));
+    // get node rect
+    const rect = document.querySelector(`[data-block-id="${nodeId}"]`)?.getBoundingClientRect();
+    if (!rect) return;
+    // set menu position
+    setStyle({
+      top: rect.top + 'px',
+      left: rect.left + 'px',
+    });
+  }, [open]);
+
+  const handleAddClick = useCallback(async () => {
+    if (!nodeId) return;
+    await insertAfter();
+  }, [nodeId, insertAfter]);
+
+  const handleDeleteClick = useCallback(async () => {
+    if (!nodeId) return;
+    await remove();
+  }, []);
```

Comment:
Do these have to be async?

Reply:
“ESLint: Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator.(@typescript-eslint/no-floating-promises)”